### PR TITLE
FactoryBot

### PR DESCRIPTION
**Pull request by stephaniewillhide**
#44

### CODE REVIEW

File: app/models/book.rb [hunk 1]
```diff
@@ -77,8 +79,8 @@ def self.per_month(year)
   private def length_of_isbn
     return if isbn.nil?
 
-    if isbn.length != 10 && isbn.length != 13
-      errors.add(:isbn, "is the wrong length (should be 10 or 13 characters)")
+    if !VALID_ISBN_LENGTHS.include? isbn.length
+      errors.add(:isbn, "is the wrong length (should be #{ VALID_ISBN_LENGTHS.to_sentence(two_words_connector: ' or ') } characters)")
```
Comment (jordanmaguire) [inline app/models/book.rb:83]: Excellent :+1:

File: spec/factories.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    name { Faker::Book.author }
+    biography { Faker::TvShows::TheFreshPrinceOfBelAir.quote }
+    publisher_name { Faker::Book.publisher }
+    publisher_email { Faker::Internet.email }
+  end
+
+  factory :genre do
+    name { Faker::Book.genre }
+  end
+end
```
Comment (jordanmaguire) [inline spec/factories.rb:33]: As this file grows, you will likely want to split these factories out into their own files for the sake of readability

EG: `spec/factories/user_factory.rb`

You don't need to do that here, yet.

File: spec/models/author_spec.rb [hunk 3]
```diff
@@ -52,13 +52,14 @@
 
   describe ".number_of_books_per_genre" do
     it "Determines how many books each Author wrote per Genre" do
-      author = Author.create!(name: "James Baldwin", biography: "James Baldwin was an American novelist, playwright, essayist, poet, and activist.", publisher_name: "Penguin Books", publisher_email: "info@penguin.com")
+      author = create(:author, name: "James Baldwin")
```
Comment (jordanmaguire) [inline spec/models/author_spec.rb:55]: Does the author name matter here?

File: spec/models/author_spec.rb [hunk 4]
```diff
@@ -52,13 +52,14 @@
...
       expect(author.number_of_books_per_genre).to eq({ })
 
-      drama = Genre.create!(name: "Drama")
-      comedy = Genre.create!(name: "Comedy")
+      drama = create(:genre, name: "Drama")
+      comedy = create(:genre, name: "Comedy")
+
+      create(:book, name: "Emma", isbn: 9292929292, created_at: Date.new(2000, 1, 1), authors: [author], genres: [drama, comedy])
```
Comment (jordanmaguire) [inline spec/models/author_spec.rb:61]: Do we need the book name, ISBN, and created_at here?

File: spec/models/author_spec.rb [hunk 5]
```diff
@@ -75,22 +76,19 @@
 
   describe ".bibliography" do
     it "Returns a human readable string containing an Author's bibliography" do
+      author = create(:author, name: "James Baldwin", biography: "James Baldwin was an American novelist, playwright, essayist, poet, and activist.", publisher_name: "Penguin Books", publisher_email: "info@penguin.com")
```
Comment (jordanmaguire) [inline spec/models/author_spec.rb:79]: Do we need the bio here?

File: spec/models/author_spec.rb [hunk 6]
```diff
@@ -75,22 +76,19 @@
...
+      comedy = create(:genre, name: "Comedy")
 
-      author = Author.create!(name: "James Baldwin", biography: "James Baldwin was an American novelist, playwright, essayist, poet, and activist.", publisher_name: "Penguin Books", publisher_email: "info@penguin.com")
-      expect(author.bibliography).to eq("")
-
-      drama = Genre.create!(name: "Drama")
-      comedy = Genre.create!(name: "Comedy")
-
-      Book.create!(name: "Emma", isbn: 9292929292, created_at: Date.new(2000, 1, 1), authors: [author], genres: [drama, comedy])
+      create(:book, name: "Emma", isbn: 9292929292, created_at: Date.new(2000, 1, 1), authors: [author], genres: [drama, comedy])
```
Comment (jordanmaguire) [inline spec/models/author_spec.rb:83]: do we need the ISBN here?

**Review by jordanmaguire — APPROVED** [review]
Looking good 👍🏻

File: spec/models/author_spec.rb (hunk 6 above)
Comment (stephaniewillhide) [reply spec/models/author_spec.rb]: Nope! Have removed all this extra stuff.

**Pull request merged**